Project-II-Calculator/calculator/src/App.js

```diff
@@ -1,23 +1,146 @@
-import React from 'react';
-import './App.css';
-
-const App = () => {
-  return (
-    <div>
-      <h3>Welcome to React Calculator</h3>
-      <p>
-        We have given you a starter project. You'll want to build out your
-        components in their respective files, remove this code and replace it
-        with the proper components.
-      </p>
-      <p>
-        <strong>
-          Don't forget to `default export` your components and import them here
-          inside of this file in order to make them work.
-        </strong>
-      </p>
-    </div>
-  );
-};
+import React from "react";
+import "./App.css";
+import Display from "./components/DisplayComponents/CalculatorDisplay";
+import WhiteButtons from "./components/ButtonComponents/NumberButton";
+import RedButtons from "./components/ButtonComponents/ActionButton";
 
+const funcs = ["÷", "x", "-", "+", "="];
+const ints = [1, 2, 3, 4, 5, 6, 7, 8, 9];
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      displayVal: "0",
+      postOp: false,
+      storedVal: null,
+      storedOp: null
+    };
+  }
+
+  clear() {
+    let disp = this.state.displayVal;
+    if (this.state.postOp) {
+      this.setState({ displayVal: "confirm clear" });
+    }
+    if (!this.state.postOp || disp === "confirm clear") {
+      this.setState({
+        displayVal: "0",
+        postOp: false,
+        postEqual: false,
+        storedVal: null,
+        storedOp: null
+      });
+    }
+  }
+
+  numInput(num) {
+    let disp = this.state.displayVal;
+    if (disp.length > 11) {
+      this.setState({ displayVal: "it done broke" });
+      return;
+    }
+    if (disp === "0" || disp === this.state.storedVal || this.state.postEqual || disp === "confirm clear") {
+      this.setState({ displayVal: String(num), postEqual:false});
+    } else {
+      this.setState({ displayVal: disp + num });
+    }
+  }
+
+  opInput(func) {
+    let disp = this.state.displayVal;
+
+    if (func === "÷") {
+      this.setState({
+        postOp: true,
+        storedVal: disp,
+        storedOp: func
+      });
+    }
+    if (func === "x") {
+      this.setState({
+        postOp: true,
+        storedVal: disp,
+        storedOp: func
+      });
+    }
+    if (func === "-") {
+      this.setState({
+        postOp: true,
+        storedVal: disp,
+        storedOp: func
+      });
+    }
+    if (func === "+") {
+      this.setState({
+        postOp: true,
+        storedVal: disp,
+        storedOp: func
+      });
+    }
+    if (func === "=") {
+      let stored = this.state.storedVal;
+      let op = this.state.storedOp;
+      let result;
+      if (op === "÷") {
+        result = parseFloat(stored) / parseFloat(disp);
+        result = result.toFixed(2);
```

Comment:
using the toFixed method like this causes two decimal places to be inserted regardless of whether they are needed. probably not the best way to display these results.